### ISSUE TITLE
Filters: Canned responses get attached to ticket, but are not send by e-mail as desired

### ISSUE THREAD

**Issue by 2kit** [osTicket/osTicket#1897]
We have set up some filters with canned responses to leverage the power of sending automated responses depending on the content that a user provided when the ticket is opened.

The filters work so far that the canned response is saved in the database ans shows up in the ticket marked as "SYSTEM (Canned Reply)"

Unfortunately the mail is never really send to the user. Mail Setup is correct, because normal mail behaviour works, also in the mail diagnostics tool.

Any ideas on what paramter or setting might be wrong?

**Comment by greezybacon** [comment]
Is the incoming mail part of a distribution list?

**Comment by 2kit** [comment]
No, its just a random personal gmail.com account. We have also verified this with other mail accounts.

**Comment by greezybacon** [comment]
@2kit Can you post the raw version of an email which triggers this issue?

**Comment by 2kit** [comment]
It is basically every email that matches the filter criteria in the mail-filter. How can I get the raw data of an e-mail out of OSTicket? Do I need to fetch them on mailserver level?

Maybe the filter settings will help you? 
We have setup 5 different filters, which are configured the same. They route the incoming mails to different departments and (should) send a canned-response in the case that one of the filter-criteria matches. See below for an example.

Filter Name: some random name
Execution Order: 1
Filter Status: Active
Target Channel: One of the available channels (we have tested all channels)
Filter Rules: Match Any
4 different matching criteria, each of them on <Ticket / Issue Details> <contains> <filter string>
Reject Ticket: No checkmark
Reply-To Email: No checkmark
Ticket auto-response: Checked
Canned Response: A response from the available list
Deparment: One of our departmens
Status: Default
Priority: Normal
SLA Plan: System Default
Auto-assign To: One of our agents
Help Topic: One of the available help topics

**Comment by 2kit** [comment]
@greezybacon: Any updates on this?

We have just found a little bug that is maybe correlated to the issue.

I set up a filter with following matching criteria: 
"Ticket / Issue Details" + "Starts With" + "mytest"

When I send a mail to the ticket system containing "mytest" at the start of the mail body, nothing happens. The filter rule does not get applied.

When I set up the same filter using "Ticket / Issue SUMMARY" and write "mytest blabla" in the email subject, then the rule gets applied and it works perfect.

It seems like there is something wrong with recognizing the content of the email being send. So far it seems like only the criteria  "Starts With" and "Ends With" are affected.

**Comment by greezybacon** [comment]
@2kit are you sending HTML email? It very likely starts with HTML tags in such a case.

**Comment by 2kit** [comment]
@greezybacon: good point! Can you tell me where I can find the content of the email body that is saved in the database? I would like to see what is actually saved. Or does the html get filtered out before saving to the database?

**Comment by greezybacon** [comment]
@2kit the filtering depends; however, it is saved in the os_ticket_thread table in the `body` column and `format` columns.

**Comment by dmnc-net** [comment]
I found this bug too accidentally (while testing ticket priorities based on rules). I've didn't received canned answer from "SYSTEM (Canned Reply)" to my email but I can see that in my web tickets. If agent uses another canned reply, I got it to my email without problems.

v1.10.1 (9ae093d)

**Comment by JediKev** [comment]
@dmnc-net are you saying that if the system sends the canned response it doesn't end in the User's mailbox but if an agent sends a canned response it does? Because I'm not seeing this on my 1.10.1 install...I get the canned response just fine from the system.

**Comment by dmnc-net** [comment]
@JediKev yes indeed, user doesn't get filter-based canned response from the system to the mailbox, just manual canned response from the agent.

IMO I suspect some problem with:
- SYSTEM agent is having problem with sending emails (default MTA is set)
- Something is wrong with "New Ticket Auto-reply" Email Template

How to debug it please?

**Comment by JediKev** [comment]
There is nothing wrong with it as I get the emails just fine. You need to look into your server error logs or mail server error logs.

**Comment by dmnc-net** [comment]
I've deleted the default template which has been created a long time ago and cloned actual stock template. I've purged filter actions as well in the same time (which was a mistake). Canned messages started to work so I set back several filter actions with no difference in the result so probably problem was caused by the old template set. Because I've already deleted that, I can't replicate this issue again :(

We can probably consider this issue as solved.
